daily-challenges/2024-09/19/lcaloi_29.spec.ts

```
@@ -0,0 +1,55 @@
+    addProduct(name: string, price: number) {
+        this.data.push({
+            name: name,
+            price: price
+        });
+    },
+    removeProduct(name: string) {
+        this.data = this.data.filter(product => product.name !== name);
+    },
+    calculateTotal() {
+        let total = 0;
+        this.data.forEach(item => total += item.price);
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)